src/modules/pivotTable/PivotTableEngine.js

```diff
@@ -498,21 +500,21 @@ export class PivotTableEngine {
     }
 
     getDependantTotalCells({ row, column }) {
-        const rowSubtotalSize = this.dimensionLookup.columns[0].size + 1
+        const rowSubtotalSize = this.dimensionLookup.columns[0]?.size + 1
```

Comment:
Just curious, how does the `?` work here (and what's `size` btw)?

Reply:
`?.` is this operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
The `?` is [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).  if `columns[0]` is undefined the expression will return undefined, otherwise it will de-reference the `.size` property.  This prevents a runtime exception.  It's similar to `(columns[0] !== null && columns[0] !== undefined) ? columns[0] : columns[0].size`

Size is the span in columns or rows of **each item** in the dimension.  Basically, if it's the "outermost" dimension it's large and if it's the "innermost" it's always 1.  There's a function called `addSize` which calculates this.

Reply:
Check the output here: https://babeljs.io/en/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBCBGArGBeGBvGBLAJgLhgEYYBfAKDNEljFTiQH4A6XAbiA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.8.7

Reply:
Thanks! Thats awesome, been waiting for something like this for ages 🚀